utils/Game.py

```diff
@@ -1,39 +1,53 @@
+from utils.ScoreLogger import ScoreLogger
 from utils.Logger import Logger
 import gym
 
 class Game:
-    def __init__(self, steps, strategy):
-        self.steps = steps
+    def __init__(self, episodes, strategy, render=False):
+        self.episodes = episodes
         self.strategy = strategy
         self.logger = Logger()
+        self.scorelogger = ScoreLogger()
+        self.render = render
 
-        self.env = gym.make('CartPole-v0')
-        self.env.reset()
+        self.env = gym.make('CartPole-v1')
 
-    def log(self, observation):
+    def log(self, score, observation):
+        self.scorelogger.log(score)
         self.logger.log(
-            self.strategy.get_pole_position(observation),
-            self.strategy.get_pole_velocity(observation),
-            self.strategy.get_pole_angle(observation),
-            self.strategy.get_pole_tip_velocity(observation)
-        )
+             self.strategy.get_pole_position(observation),
+             self.strategy.get_pole_velocity(observation),
+             self.strategy.get_pole_angle(observation),
+             self.strategy.get_pole_tip_velocity(observation)
+         )
 
     def print_logs(self):
-        print(self.logger.get_positions())
-        print(self.logger.get_velocities())
-        print(self.logger.get_angles())
-        print(self.logger.get_tip_velocities())
+         print(self.logger.get_positions())
+         print(self.logger.get_velocities())
+         print(self.logger.get_angles())
+         print(self.logger.get_tip_velocities())
+         print(self.scorelogger.get_scores())
 
     def plot(self, title=""):
-        self.logger.plot(title)
+        self.scorelogger.plot(title)
```

Comment:
would it be possible to join the two plots?

Reply:
It’s possible but since the plotting before was based on time steps and the plotting now is based on 100 episodes running it would produce 100 graphs then for each strategy.